### PR TITLE
feat: rpc endpoint to list addresses

### LINKED ISSUE #1250: New API Endpoint Needed
We definitely would love an RPC API endpoint added that returns all the Ethereum addresses on the network.

### PR DESCRIPTION
**Pull request by zachdaniel**
Resolves #1250

### Enhancements
* Add an RPC endpoint that returns addresses, paginated, ordered by inserted_at

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex [hunk 1]
```diff
@@ -260,6 +278,13 @@ defmodule BlockScoutWeb.API.RPC.AddressController do
     Enum.any?(address_hashes, &(&1 == :error))
   end
 
+  defp list_accounts(%{page_number: page_number, page_size: page_size}) do
+    offset = (max(page_number, 1) - 1) * page_size
+
+    # limit is just page_size
```
Comment (goodsoft) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex]: If this number can be provided by client, we should probably define some reasonable limit.

Comment (zachdaniel) [reply apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex]: The maximum page size is capped already, here: https://github.com/poanetwork/blockscout/pull/1589/files#diff-b6f6ac579ae480874463adb5891b5e40L174

Comment (zachdaniel) [reply apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex]: It is capped at 10,000

File: apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex [hunk 2]
```diff
@@ -422,7 +447,10 @@ defmodule BlockScoutWeb.API.RPC.AddressController do
     end
   end
 
-  defp get_token_balance(%{contract_address_hash: contract_address_hash, address_hash: address_hash}) do
+  defp get_token_balance(%{
```
Comment (ayrat555) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex]: is this change necessary?

Comment (zachdaniel) [reply apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex]: No, for some reason my formatter changed these lines. I'll fix them.
